debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSnapshotChangeEventSource.java

```diff
@@ -191,8 +191,10 @@ protected void lockTablesForSchemaSnapshot(ChangeEventSourceContext sourceContex
                 // Continue anyway, since RDS (among others) don't allow setting a global lock
                 assert !isGloballyLocked();
             }
-            // FLUSH TABLES resets TX and isolation level
-            connection.executeWithoutCommitting("SET TRANSACTION ISOLATION LEVEL REPEATABLE READ");
+            if (!connectorConfig.getSnapshotLockingMode().equals(MySqlConnectorConfig.SnapshotLockingMode.MINIMAL_PERCONA)) {
```

Comment:
Instead of checking the particular enum constant, can we have a more expressive method on the enum type itself? Like `boolean flushResetsIsolationLevel()` or something?